.github/workflows/e2e-tests.yml

```diff
@@ -99,16 +101,17 @@ jobs:
             - name: Cypress run
               uses: cypress-io/github-action@v6
               with:
-                  record: true
-                  parallel: true
-                  group: e2e-chrome-parallel-${{ matrix.versions }}
+                  record: ${{ env.SHOULD_RECORD }}
+                  parallel: ${{ env.SHOULD_RECORD }}
```

Comment:
What is the reason for making the parallel setting depending on `SHOULD_RECORD`? Currently running the tests takes around 10 minutes, with this change it takes around 40 minutes to complete. Does not running in parallel save a lot of Cypress Cloud credits? Thank you!

Reply:
Thank you Simona for the great question! 
The parallel setting is dependent on record because Cypress only supports parallel execution when tests are being recorded in their cloud. Before my changes, tests ran in parallel on Cypress's infrastructure. After the changes, they still run in parallel but within our own CI infrastructure. It appears we might need to review our CI configurations to ensure we have enough resources and workers for efficient parallel execution, which could help address the increased test durations.

Reply:
Thanks for your insightful question, Simona!

Regarding the parallel configuration, it’s linked to the `SHOULD_RECORD` setting because Cypress Cloud requires `parallel` execution to be tied to `record`.

Initially, I thought our tests were running in parallel on GitHub Actions, but they were actually executing multiple times simultaneously without true parallelism.

To correct this, I've implemented specific test chunks through  [generateTestMatrix](https://github.com/dhis2/capture-app/blob/a602899bc4698e17a32eab7554781cfb03bdcaf0/cypress/support/generateTestMatrix.js) script and increased the parallel runs from 5 to 10. This adjustment not only resolves the parallelisation issue but also optimises our test execution time, making it even more efficient than on Cypress Cloud.

Please review the updated setup, and I’m here to discuss any part of it or share further insights into the changes I've made!